lib/postgrex/connection.ex

```diff
@@ -167,16 +134,11 @@ defmodule Postgrex.Connection do
 
     * `:timeout` - Call timeout (default: `#{@timeout}`)
   """
-  @spec listen(pid, String.t, Keyword.t) :: {:ok, reference} | {:error, Postgrex.Error.t}
+  @spec listen(pid, String.t, Keyword.t) :: {:ok, reference}
```

Comment:
On this branch listen/unlisten will crash the Connection if there is an error. Note that on master the behaviour is still ambiguous on an error, #111 was not a good enough fix :(.